branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -29,6 +30,28 @@ private static List<IBranchLayoutEntry> collectEntriesRecursively(IBranchLayoutE
     return entryByName.get(branchName).getOrNull();
   }
 
+  @Override
+  public @Nullable IBranchLayoutEntry findNextEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > -1 && currentIndex + 1 < entriesOrderedList.length()) {
+      @LTLengthOf("entriesOrderedList") int nextIndex = currentIndex + 1;
+      return findEntryByName(entriesOrderedList.get(nextIndex));
+    }
+    return null;
+  }
+
+  @Override
+  public @Nullable IBranchLayoutEntry findPreviousEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > 0 && currentIndex < /* added for preventing a type error */entriesOrderedList.length() - 1) {
+      @LTLengthOf("entriesOrderedList") int previousIndex = currentIndex - 1;
+      return findEntryByName(entriesOrderedList.get(previousIndex));
+    }
+    return null;
+  }
```

Comment:
```suggestion
  @Override
  public @Nullable IBranchLayoutEntry findNextEntry(String branchName) {
    return findEntry(branchName, -1, entriesOrderedList -> entriesOrderedList.length() - 1, /* shift */ 1);
  }

  @Override
  public @Nullable IBranchLayoutEntry findPreviousEntry(String branchName) {
    return findEntry(branchName, 0, List::length, /* shift */ -1);
  }

  private @Nullable IBranchLayoutEntry findEntry(String branchName, int lowerBound, Function1<List<String>, Integer> upperBound, int shift) {
    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
    val currentIndex = entriesOrderedList.indexOf(branchName);
    if (currentIndex > lowerBound && currentIndex < upperBound.apply(entriesOrderedList)) {
      @LTLengthOf("entriesOrderedList") int index = currentIndex + shift;
      return findEntryByName(entriesOrderedList.get(index));
    }
    return null;
  }
```

now, probably I am overthinking 😆 but it's a few lines shorter!

Reply:
hmmm, that's a nice refactoring indeed 👍 I was just wondering about the use cases where we need to jump more than one branch up or down🤔 any ideas?